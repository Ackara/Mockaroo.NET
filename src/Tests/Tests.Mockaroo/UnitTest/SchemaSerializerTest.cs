using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Gigobyte.Mockaroo;
using Gigobyte.Mockaroo.Fields;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.IO;

namespace Tests.Mockaroo.UnitTest
{
    [TestClass]
    [DeploymentItem(SampleData.DirectoryName)]
    [UseApprovalSubdirectory(nameof(ApprovalTests))]
    [UseReporter(typeof(DiffReporter), typeof(ClipboardReporter))]
    public class SchemaSerializerTest
    {
        [ClassCleanup]
        public static void Cleanup()
        {
            ApprovalTests.Maintenance.ApprovalMaintenance.CleanUpAbandonedFiles();
        }

        [TestMethod]
        [Owner(Dev.Ackara)]
        public void WriteObject_should_serialize_a_schema_object_into_json()
        {
            // Arrange
            string serializedData;
            var sut = SampleData.CreateSchema();

            // Act
            var data = sut.Serialize();
            using (var reader = new StreamReader(data))
            { serializedData = reader.ReadToEnd(); }

            // Assert
            Approvals.VerifyJson(serializedData);
        }

        [TestMethod]
        [Owner(Dev.Ackara)]
        public void ReadObject_should_create_a_schema_object_from_a_stream_of_json()
        {
            // Arrange
            var sut = new Schema();
            var sampleFile = SampleData.GetFile(Asset.SchemaJson);

            // Act
            sut.Deserialize(sampleFile.OpenRead());

            // Assert
            Assert.AreEqual(3, sut.Count);
            Assert.AreEqual(DataType.Words, sut[0].Type);
            Assert.IsInstanceOfType(sut[1], typeof(NumberField));
        }
    }
}