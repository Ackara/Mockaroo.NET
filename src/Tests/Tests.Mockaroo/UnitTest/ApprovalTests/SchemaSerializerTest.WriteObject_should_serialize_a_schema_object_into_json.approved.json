[
    {
        "name": "Title",
        "type": "Words",
        "percentageBlank": "0",
        "formula": "",
        "min": "3",
        "max": "5"
    },
    {
        "name": "Views",
        "type": "Number",
        "percentageBlank": "0",
        "formula": "",
        "min": "3",
        "max": "1000",
        "decimals": "0"
    },
    {
        "name": "PostDate",
        "type": "Date",
        "percentageBlank": "0",
        "formula": "",
        "min": "01/01/2000",
        "max": "01/01/2010"
    }
]